src/Agent.Worker/Build/ArtifactCommandExtension.cs

```diff
@@ -234,6 +320,50 @@ private void ProcessArtifactUploadCommand(IExecutionContext context, Dictionary<
             var artifact = await buildHelper.AssociateArtifact(buildId, name, ArtifactResourceTypes.Container, fileContainerFullPath, propertiesDictionary, cancellationToken);
             context.Output(StringUtil.Loc("AssociateArtifactWithBuild", artifact.Id, buildId));
         }
+		
+        private async Task UploadDropArtifactAsync(
```

Comment:
@TingluoHuang This code needs to move to a out of proc plugin for stability of the agent, right?

Reply:
yes.